File: classes02/python/desaturation.py
import cv2
import numpy as np

img=cv2.imread("../data/images/capsicum.jpg")
saturationScale=0.1
hsvImage=cv2.cvtColor(img,cv2.COLOR_BGR2HSV)

hsvImage=np.float32(hsvImage)

H,S,V=cv2.split(hsvImage)
S=np.clip(S*saturationScale,0,255)

hsvImage=np.uint8(cv2.merge([H,S,V]))
imSat=cv2.cvtColor(hsvImage,cv2.COLOR_HSV2BGR)

combined=np.hstack([img,imSat])
cv2.imshow("Combined",combined)
cv2.waitKey(0)
cv2.destroyAllWindows()

File: classes02/cpp/desaturation.cpp
#include <opencv2/opencv.hpp>
#include <math.h>

using namespace cv;
using namespace std;

int main(void)
{
    Mat img=imread("../data/images/capsicum.jpg");

    float staturationScale=0.01;

    Mat hsvImage;
    cvtColor(img,hsvImage,COLOR_BGR2HSV);

    hsvImage.convertTo(hsvImage,CV_32F);

    vector<Mat> channels(3);
    split(hsvImage,channels);

    channels[1]=channels[1]*staturationScale;

    min(channels[1],255,channels[1]);
    max(channels[1],0,channels[1]);

    merge(channels,hsvImage);

    hsvImage.convertTo(hsvImage, CV_8UC3);

    Mat imSat;
    cvtColor(hsvImage, imSat, COLOR_HSV2BGR);

    Mat combined;
    hconcat(img,imSat,combined);

    imshow("Combined", combined);

    waitKey(0);
    destroyAllWindows();


    return EXIT_SUCCESS;
}